src/model/event.rs

```diff
@@ -406,22 +407,29 @@ pub struct MessageDeleteEvent {
 #[non_exhaustive]
 pub struct MessageUpdateEvent {
     pub id: MessageId,
-    pub guild_id: Option<GuildId>,
     pub channel_id: ChannelId,
-    pub kind: Option<MessageType>,
+    pub author: Option<User>, // TODO: Is this a Message field that can even change?
     pub content: Option<String>,
-    pub nonce: Option<String>,
-    pub tts: Option<bool>,
-    pub pinned: Option<bool>,
-    pub timestamp: Option<Timestamp>,
+    pub timestamp: Option<Timestamp>, // TODO: Is this a Message field that can even change?
     pub edited_timestamp: Option<Timestamp>,
-    pub author: Option<User>,
+    pub tts: Option<bool>,
     pub mention_everyone: Option<bool>,
     pub mentions: Option<Vec<User>>,
     pub mention_roles: Option<Vec<RoleId>>,
+    pub mention_channels: Option<Vec<ChannelMention>>,
     pub attachments: Option<Vec<Attachment>>,
     pub embeds: Option<Vec<Embed>>,
+    pub reactions: Option<Vec<MessageReaction>>,
+    pub nonce: Option<String>, // TODO: Is this a Message field that can even change?
+    pub pinned: Option<bool>,
+    pub kind: Option<MessageType>, // TODO: Is this a Message field that can even change?
     pub flags: Option<MessageFlags>,
+    pub components: Option<Vec<ActionRow>>,
+    #[deprecated(note = "deprecated by Discord")]
+    pub stickers: Option<Vec<StickerItem>>,
+    pub sticker_items: Option<Vec<StickerItem>>,
+
+    pub guild_id: Option<GuildId>, // TODO: Is this a Message field that can even change?
```

Comment:
Regarding these `TODO` comments, I do not think so? They are all fields that are filled in when the message is created, and should stay the way they are. Unless Discord adds the ability to, say, move messages from one place to another, which could change fields like `guild_id`. But that's unlikely to be added as a feature.

We can leave them in, though, for pedantic purposes (maybe Discord *could* go crazy and allow changing those fields). I would just remove the comments; they're redundant.

Reply:
> Regarding these TODO comments, I do not think so? They are all fields that are filled in when the message is created, and should stay the way they are. Unless Discord adds the ability to, say, move messages from one place to another, which could change fields like guild_id. But that's unlikely to be added as a feature.

I agree

> We can leave them in, though, for pedantic purposes (maybe Discord could go crazy and allow changing those fields).

If so, we should pull it through and add all of Message's fields. The reason I haven't done that is because then, for consistency, we should also process all those other fields in the CacheUpdate implementation, which is non-trivial. Also, it blows up the size of the type, and is potentially confusing to users who think those properties of a Message can actually change. What do you think? I can also open another PR to remove the author, timestamp, kind... fields on next.

Reply:
> Also, it blows up the size of the type, and is potentially confusing to users who think those properties of a Message can actually change. 

Yeah, when I was making that comment, I thought of the size of the struct that would get ridiculously large if we add all of the fields. If the fields can't be altered, perhaps it would just be best to remove them altogether. They can be re-added at a later date if Discord allows it in the future. You can remove them on `next`